### PR TITLE
Fix #6714

### LINKED ISSUE #6714: Show output in wtf command if it takes more than N seconds to finish

### PR DESCRIPTION
**Pull request by prashast**
This fix notifies the user if the write to a file(using wtf!) is taking more than N seconds, right now I've hard-coded it to 3 seconds. This is my first pull request so let me know if there are any coding guidelines that I'm not adhering to or if the current implementation has some shortcomings.

### CODE REVIEW

**Comment by prashast** [comment]
Aah, maybe I misunderstood the N seconds mentioned in the issue. Should I modify the command to something like `wtf [filename] [size] [time]` and print message if it's taking more than `time` seconds to finish?

File: libr/core/cio.c [hunk 1]
```diff
@@ -86,6 +89,13 @@ R_API int r_core_dump(RCore *core, const char *file, ut64 addr, ut64 size, int a
 	}
 	r_cons_break_push (NULL, NULL);
 	for (i = 0; i < size; i += bs) {
+		if (check_time) {
+		elapse = time (NULL) - start;
+		if (elapse > 3) {
```
Comment (radare) [inline libr/core/cio.c]: Wrong indent

File: libr/core/cio.c [hunk 2]
```diff
@@ -63,8 +63,11 @@ R_API int r_core_seek_base (RCore *core, const char *hex) {
 R_API int r_core_dump(RCore *core, const char *file, ut64 addr, ut64 size, int append) {
 	ut64 i;
 	ut8 *buf;
-	int bs = core->blocksize;
+	int bs = core->blocksize,check_time=1;
```
Comment (radare) [inline libr/core/cio.c]: Use bool

File: libr/core/cio.c [hunk 3]
```diff
@@ -63,8 +63,11 @@ R_API int r_core_seek_base (RCore *core, const char *hex) {
...
 	FILE *fd;
+	time_t start,elapse;
+
+	start = time (NULL);
```
Comment (radare) [inline libr/core/cio.c]: This is not portable and slow

File: libr/core/cio.c [hunk 4]
```diff
@@ -86,6 +89,13 @@ R_API int r_core_dump(RCore *core, const char *file, ut64 addr, ut64 size, int a
 	}
 	r_cons_break_push (NULL, NULL);
 	for (i = 0; i < size; i += bs) {
+		if (check_time) {
+		elapse = time (NULL) - start;
```
Comment (radare) [inline libr/core/cio.c]: This is slow

**Comment by radare** [comment]
We use r_sys_now() which is portable. Also checking for time on each iter is slow

You may consider checking how its done in / for searcxhing to show some feedback on the eta and such

File: libr/core/cio.c (hunk 2 above)
Comment (XVilka) [inline libr/core/cio.c]: moreover, use right indentation.

**Pull request closed without merging**